### PR TITLE
feat(react-dialog): implements DialogContent as a swap of DialogBody

### PR DESCRIPTION
**Pull request by bsunderhus**
## Current Behavior

Currently `DialogSurface` has a grid style that requires its child elements to be a combination of `DialogTitle`, `DialogBody` and `DialogActions`.

## New Behavior

Creates a separation layer between `DialogSurface` and the proper grid to allow custom semantic elements to be positioned between the dialog content and the grid itself without breaking the grid, a case for this to be required is: `<form>`:

```tsx
export const Formulary = () => {
  const styles = useStyles();
  const handleSubmit = (ev: React.FormEvent) => {
    ev.preventDefault();
    alert('form submitted!');
  };
  return (
    <Dialog modalType="non-modal">
      <DialogTrigger>
        <Button>Open formulary dialog</Button>
      </DialogTrigger>
      <DialogSurface>
        <form onSubmit={handleSubmit}>
          <DialogBody>
            <DialogTitle>Dialog title</DialogTitle>
            <DialogContent className={styles.content}>
              <Label required htmlFor={'email-input'}>
                Email input
              </Label>
              <Input required type="email" id={'email-input'} />
              <Label required htmlFor={'password-input'}>
                Password input
              </Label>
              <Input required type="password" id={'password-input'} />
            </DialogContent>
            <DialogActions>
              <DialogTrigger>
                <Button appearance="secondary">Close</Button>
              </DialogTrigger>
              <Button type="submit" appearance="primary">
                Submit
              </Button>
            </DialogActions>
          </DialogBody>
        </form>
      </DialogSurface>
    </Dialog>
  );
};
```

Also:

1. Updates context ID to Id
2. Adds `type="button"` to `DialogTrigger` child props, to ensure behaviour on formularies

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                          |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                      Change |
| :------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr> | `81.199 kB`<br />`24.196 kB` | `82.755 kB`<br />`24.581 kB` | `1.556 kB` <br />`385 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `188.672 kB`<br />`52.359 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.394 kB`<br />`11.007 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.851 kB`<br />`1.964 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/8e656f25a3867cefcb60c0ae6e156fee3a674277'>8e656f25a3867cefcb60c0ae6e156fee3a674277</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/Avatar-mount_base.html">1549</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/Avatar-mount.html">1559</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/Button-mount_base.html">1111</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/Button-mount.html">1182</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/FluentProvider-mount_base.html">1830</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/FluentProvider-mount.html">1847</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/FluentProviderWithTheme-mount_base.html">760</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/FluentProviderWithTheme-mount.html">768</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">696</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">686</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">775</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">740</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/MakeStyles-mount_base.html">2240</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/MakeStyles-mount.html">2304</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/SpinButton-mount_base.html">2997</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24855/perf-test-react-components/SpinButton-mount.html">2982</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts [hunk 1]
```diff
@@ -34,8 +34,8 @@ export const useDialogSurface_unstable = (
   const dialogRef = useDialogContext_unstable(ctx => ctx.dialogRef);
   const open = useDialogContext_unstable(ctx => ctx.open);
   const requestOpenChange = useDialogContext_unstable(ctx => ctx.requestOpenChange);
-  const dialogTitleID = useDialogContext_unstable(ctx => ctx.dialogTitleID);
-  const dialogBodyID = useDialogContext_unstable(ctx => ctx.dialogBodyID);
+  const dialogTitleID = useDialogContext_unstable(ctx => ctx.dialogTitleId);
+  const dialogBodyID = useDialogContext_unstable(ctx => ctx.dialogContentId);
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts:38]:
```suggestion
  const dialogContentID = useDialogContext_unstable(ctx => ctx.dialogContentId);
```

File: packages/react-components/react-dialog/src/components/DialogTrigger/useDialogTrigger.ts [hunk 2]
```diff
@@ -39,6 +39,7 @@ export const useDialogTrigger_unstable = (props: DialogTriggerProps): DialogTrig
     children: applyTriggerPropsToChildren<DialogTriggerChildProps>(
       children,
       useARIAButtonProps(child?.type === 'button' || child?.type === 'a' ? child.type : 'div', {
+        type: 'button',
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/components/DialogTrigger/useDialogTrigger.ts:42]: shouldn't `useARIAButton` set the type by default ?

File: packages/react-components/react-dialog/src/stories/Dialog/DialogFormulary.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+const useStyles = makeStyles({
+  content: {
+    display: 'flex',
+    flexDirection: 'column',
+    rowGap: '10px',
+  },
+});
+
+export const Formulary = () => {
```
Comment (ling1726) [inline packages/react-components/react-dialog/src/stories/Dialog/DialogFormulary.stories.tsx:22]: `AsForm`, `WithForm`, `FormDialog`

Formulary makes no sense

File: packages/react-components/react-dialog/src/components/DialogTrigger/useDialogTrigger.ts (hunk 2 above)
Comment (bsunderhus) [reply packages/react-components/react-dialog/src/components/DialogTrigger/useDialogTrigger.ts:42]: That's a valid point, I'll open an issue to update `useARIAButtonProps` on that.

**Review by ling1726 — APPROVED** [review]

**Pull request merged**